jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -386,10 +434,17 @@ public static class ConcurrentEntry<E> implements Entry<E>
         // Other threads accessing must check the state field above first, so a good before/after
         // relationship exists to make a memory barrier.
         private E pooled;
+        private final Holder<E> holder;
 
         public ConcurrentEntry(ConcurrentPool<E> pool)
         {
             this.pool = pool;
+            holder = pool.weak ? new WeakHolder<>(this) : new StrongHolder<>(this);
+        }
+
+        private Holder<E> getHolder()
```

Comment:
I'd just reference the final field and remove the method.

Reply:
why?